spec/controllers/coronavirus_landing_page_controller_spec.rb

```diff
@@ -25,5 +25,54 @@
         expect(response.headers["Cache-Control"]).to eq("max-age=#{30.seconds}, public")
       end
     end
+
+    context "when testing national_applicability" do
+      render_views
+
+      before do
+        allow(Rails.env).to receive(:production?).and_return(true)
+      end
+
+      it "loads the production content item in production environments" do
+        expect(CoronavirusTimelineNationsContentItem).to_not receive(:load)
+
+        get :show, params: { timeline_nation: "foo" }
+      end
+
+      it "shows no content text when there are no timeline entries for a nation" do
+        stub_content_store_has_item("/coronavirus", coronavirus_content_item_with_timeline_national_applicability_without_wales)
+        get :show, params: { timeline_nation: "wales" }
+
+        expect(response.body).to have_content("There haven't been any recent updates for Wales.")
+      end
+
+      it "shows timeline for England" do
+        stub_content_store_has_item("/coronavirus", coronavirus_content_item_with_timeline_national_applicability)
+        get :show, params: { timeline_nation: "england" }
+
+        expect(response.body).to have_content("In England")
```

Comment:
This test will pass for any param values of timeline_nation, if I understand right, as we'll always have this content, the determining factor will be what class the selector has. E.g. for England this would be `div#nation-england:not(.covid-timeline__item--hidden)`.

Might be good to have a second assertion that we only have England selected with an assertion of `expect(response.body).not_to have_selector("div.covid-timeline__item:not(.covid-timeline__item--hidden).not(#nation-england)")` though it's a pretty fierce selector (also requires `covid-timeline__item` as per my previous comment)

Reply:
This works too:

```
expect(response.body).to have_selector(".js-covid-timeline", text: "In England")
expect(response.body).to have_selector(".js-covid-timeline", text: "In Wales", visible: false)
```

Reply:
As it's a controller spec it's not rendering the content so it doesn't know if the content is visible or not (it's unfortunate it doesn't throw an error).

For example in this test:

```
(byebug) expect(response.body).to have_content("In England")
true
(byebug) expect(response.body).to have_content("In Wales")
true
(byebug) expect(response.body).to have_selector(".js-covid-timeline", text: "In England", visible: false)
true
(byebug) expect(response.body).to have_selector(".js-covid-timeline", text: "In England", visible: true)
true
(byebug) expect(response.body).to have_selector(".js-covid-timeline", text: "In Wales", visible: true)
true
(byebug) expect(response.body).to have_selector(".js-covid-timeline", text: "In Wales", visible: false)
true
```

Reply:
It feels like the controller tests and non-JS feature tests are suffering from the same problems. It would be great if we could figure out a nice, readable, way to test for the absence of content that applies to both.

cc @andysellick